packages/build/src/core/types.ts

```diff
@@ -2,7 +2,7 @@ import { NetlifyConfig } from '../../types/index.js'
 
 export type Mode = 'buildbot' | 'cli' | 'require'
 
-export type BuildCLIFlags = {
+export interface BuildCLIFlags {
```

Comment:
Is there a reason why you convert it to an interface?

Types have the big advantage that you can build easy joined types.

```typescript
interface Otherthing extends BuildCLIFlags, OtherType {}

// vs

type Otherthing = BuildCLIFlags & Omit<OtherType, 'someProp'>
```

So the `type` offers more flexibility

Reply:
I usually just follow what the[ TS docs ](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces)mention:

> For the most part, you can choose based on personal preference, and TypeScript will tell you if it needs something to be the other kind of declaration. If you would like a heuristic, use interface until you need to use features from type.

but reverted